eng/pipelines/templates/stages/archetype-spec-gen-sdk.yml

```diff
@@ -0,0 +1,263 @@
+                Write-Host "##vso[task.logissue type=error]One or more required variables is empty or invalid. Ensure that SpecRepoUrl and SdkRepoUrl are set to valid GitHub repository URLs."
+                Exit 1
+              }
+
+            displayName: "Create Run Time Variables"
+
+          - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+            parameters:
+              Paths:
+              - '/*'
+              - '!sdk/**/test-recordings/*'
+              - '!sdk/**/recordings/*'
+              - '!sdk/**/SessionRecords/*'
+              - '!sdk/**/session-records/*'
+              Repositories:
+              - Name: $(SpecRepoOwner)/$(SpecRepoName)
+                Commitish: ${{ parameters.SpecRepoCommit }}
+                WorkingDirectory: $(SpecRepoDirectory)
+              - Name: $(SdkRepoOwner)/$(SdkRepoName)
+                Commitish: ${{ parameters.SdkRepoCommit }}
+                WorkingDirectory: $(SdkRepoDirectory)
+              SkipCheckoutNone: true
+
+          - script: |
+              if [ "${{ parameters.SpecRepoCommit }}" = "$(Build.SourceBranch)" ]; then
+                cd $(SpecRepoDirectory)
+                default_commit=$(git rev-parse HEAD)
+                echo "##vso[task.setvariable variable=SpecRepoCommit]$default_commit"
+                echo "SpecRepoCommit variable set to default commit: $default_commit"
+              else
+                echo "##vso[task.setvariable variable=SpecRepoCommit]${{ parameters.SpecRepoCommit }}"
+                echo "SpecRepoCommit variable set to: ${{ parameters.SpecRepoCommit }}"
+              fi
+            displayName: 'Set SpecRepoCommit variable'
+
+          - task: NodeTool@0
+            inputs:
+              versionSpec: $(NodeVersion)
+            displayName: 'Install Node.js'
+
+          - task: UsePythonVersion@0
+            inputs:
+              versionSpec: $(PythonVersion)
+
+          - script: |
+              npm install -g @azure-tools/spec-gen-sdk@$(SpecGenSdkVersion)
+            displayName: 'Install spec-gen-sdk'
+
+          - script: |
+              optional_params=""
+              sdk_gen_info="sdk generation from Config : "
+
+              if [ "${{ parameters.ConfigType }}" = "TypeSpec" ]; then
+                optional_params="$optional_params --tsp-config-relative-path ${{ parameters.ConfigPath }}"
+                sdk_gen_info="$sdk_gen_info '${{ parameters.ConfigPath }}',"
+              else
+                optional_params="$optional_params --readme-relative-path ${{ parameters.ConfigPath }}"
+                sdk_gen_info="$sdk_gen_info '${{ parameters.ConfigPath }}',"
+              fi
+
+              if [ "$(Build.Reason)" = "PullRequest" ]; then
+                optional_params="$optional_params --pr-number=$(System.PullRequest.PullRequestNumber)"
```

Comment:
I assume you plan to have follow-up change to switch PRs over to using the runner to discover the projects instead of doing that in the spec-gen-sdk tool directly.

Reply:
Yes, those changes will be in a separate PR.